packages/addons/service/mariadb/source/bin/mariadb.start

```diff
@@ -0,0 +1,64 @@
+#!/bin/sh
+
+# SPDX-License-Identifier: GPL-2.0-or-later
+# Copyright (C) 2018-present Team LibreELEC (https://libreelec.tv)
+
+. /etc/profile
+
+MYSQL_ROOT_PASS=123
```

Comment:
could we add this to the settings ? a preset password is always rather meh

Reply:
have a look at the docker add-on, there's an approach for setting the root pass

Reply:
You mean like this? https://github.com/LibreELEC/LibreELEC.tv/blob/master/packages/addons/service/docker/source/examples/mysql.service#L19-L21
It is still hardcoded (but I will replace 123 with libreelec). User can change password from command line with mysqladmin

Reply:
what about `MYSQL_ROOT_PASS=$MYSQL_PW`

then add `settings.xml` (not tested)
```
<?xml version="1.0" encoding="utf-8" standalone="yes"?>
<settings>
    <category label="30000">
		<setting label="1010" type="lsep"/>
		<setting type="sep" />
		<setting id="MYSQL_PW" type="text" label="30001" default="" />
    </category>
</settings>
```

Reply:
Well, it is used when addon is installed. Which means initial password would always be hardcoded. And when changing it (I think) it is easier to type it on real keyboard than in kodi's virtual one.

Reply:
but you assume that ppl know where to change the pw - I don't think most of the users have a clue about that :) included at the settings won't prevent setting it at the console (i guess ? be gentle not sure how this works at mysql)

Preset the pw is meh not sure we can randomise it at all without creating a mess
maybe initial pw is "date" or something ? Just fishing for ideas to "fake" some randomness.

Reply:
I guess we need to support this ootb, something to remember.

Reply:
Personally, I agree with vPeter: person, using/installing MySQL + advancesettings.xml + clients **should know**, where db is stored, whats the password, acessibility etc. (not good idea to install "mysql for dummies" on default Rpi + SD Card configuration for example, as well using MySQL reached by internet without firewall), so it should be fine with setup password by editing files, which most propably will edit anyway.

On the other hand I understand security concerns. I think random generated pass is good idea (only-Kodi usage doesnt require root access there in ideal case anyway, as kodi user is prepared). I dont think that LE box exposed to the internet is typical/majority case, but it couldnt hurt - better safe then sorry.

I noticed, that recent ssh passwd change ability using pop up window for setup password, and I think also [maridb docker](https://github.com/linuxserver/libreelec-addon-repo/tree/master/docker.linuxserver.mariadb), maybe it could be used the same approach by copy and paste?

I think all three options have plus / minuses, and I am fine with all 3 of them :)

Reply:
As a design principle we shouldn't allow static passwords. I'd also advocate we aim for a simple user experience. Not a one-button "for dummies" but reduced complexity through sensible and appropriate GUI controls. Kodi also needs work, but that's a different battle. 

My $0.02: Passwords should be stored in the settings.xml for the add-on. If this file pre-exists when the add-on is first installed; read and reuse the root/user password values. Else create a random root password and store it. Provide a simple GUI for setting the credential to use with Kodi. Prompt for username but pre-define "kodi" so that cumbersome on-screen input is not required. Prompt for password but require input. If users choose to set "kodi" as the obvious and easily guessed password it's their mistake not ours.

IP access to the DB is partially covered with the default firewall. This feature currently defaults to OFF but will/should be changed to HOME before 9.0 release. HOME allows connections from RFC1918 networks. It won't protect the small number of idiots who expose their HTPC through deliberate port forwarding, but will protect the larger group of ignorants who expose HTPC services through a VPN connection.

Reply:
NB: Rome wasn't built in a day so perhaps start simple (set a default root user credential only and let users create a user and bad passwords themselves) and add the nicer GUI stuff in a future update.

Reply:
I'm doing some changes to create passwords on first start.

Reply:
What about something like that? https://pasteboard.co/HU1hHVg.jpg
When password is changed service is restarted and passwords enforced with running some init sql commands.
Hardcoded defaults are set to 123/kodi: https://pasteboard.co/HU1jmoV.jpg

Reply:
GUI looks great (pictures are always worth many words!) but I don't see need for hardcoded defaults. If the GUI can handle either password being changed after first-run just create two random passwords for root/kodi and we don't need to do more. Minor nit; the left side `Password` button should be `Passwords` as there's more than one. I'll drop into IRC later - ping me if I'm misunderstanding anything.

Reply:
second nit; the usernames should be root/kodi not Root/Kodi